src/de/hasait/cipa/activity/CipaFileResourceCleanup.groovy

```diff
@@ -0,0 +1,49 @@
+package de.hasait.cipa.activity
+
+import de.hasait.cipa.Cipa
+import de.hasait.cipa.CipaNode
+import de.hasait.cipa.jobprops.JobParameterContainer
+import de.hasait.cipa.jobprops.JobParameterContribution
+import de.hasait.cipa.jobprops.JobParameterValues
+import de.hasait.cipa.resource.CipaFileResource
+import de.hasait.cipa.resource.NodeCleanup
+
+/**
+ * Performs clean up of all file resources for parallel activities with respect to each node.
+ */
+class CipaFileResourceCleanup extends AbstractCipaBean implements NodeCleanup, JobParameterContribution {
+
+    static final String PARAM___CLEANUP_RESOURCES = "CLEANUP_FILE_RESOURCES"
+
+    private boolean cleanUp
+
+    CipaFileResourceCleanup(Cipa cipa) {
+        super(cipa)
+    }
+
+    @Override
+    void cleanupNode(CipaNode node) {
+        script.echo("Initializing clean up of ${node.getLabel()} for host ${node.runtimeHostname}")
+        if (cleanUp){
+            Set<CipaFileResource> cleanupFileResources = cipa.findBeans(CipaFileResource.class)
+            script.echo("Cleaning up ${cleanupFileResources.size()} file resources on node ${node.runtimeHostname}...")
```

Comment:
The size is misleading as filtering comes later and reduces the number of handled resources - I would not print it:
Cleaning up file resources on host ${node.runtimeHostname}